src/Fixer/FunctionNotation/NativeFunctionInvocationFixer.php

```diff
@@ -378,6 +378,7 @@ private function getAllCompilerOptimizedFunctionsNormalized(): array
             'is_string',
             'ord',
             'sizeof',
+            'sprintf',
```

Comment:
I know there wasn't such a convention before, but wouldn't it make sense to add `sprintf` only if PHP >= 8.4? For previous versions it does not add value, so probably people running older PHP versions not necessarily want to apply potentially huge diff that does not improve performance for them?

Reply:
I do not support this idea, as running PHP CS Fixer in different environments is even worse - if not detected from composer.json of course.

Reply:
@mvorisek I don't understand what you said. The suggested usage of Fixer is to run it on the PHP version you use for your app. So if the app is still on 7.4 for example, there's no point in applying `\` to `sprintf` as it does not bring the performance boost, but can lead to sudden CI crashes when Fixer is updated.

Reply:
I mean if we condition it on executable PHP version, I do not want different fixer output locally vs. CI when the minor PHP version does not match.

Reply:
@Wirone open-source package will need to support multiple versions of PHP. They don't have a single target PHP version. And they should be fully-qualifying calls to `sprintf` to benefit the subset of their users running on PHP >=8.4 even when their min target version is lower than 8.4 (qualifying the call does not break code on PHP 8.3. It simply does not unlock a performance optimization there)

Reply:
@mvorisek but there _are_ rules that behave differently depending on PHP runtime, that's why I wanted to discuss it.

@stof yeah, for packages it's different, but still it'll affect codebases with pinned PHP versions. I don't want to enforce this change, just wanted to figure out the best approach.

Reply:
> @mvorisek but there _are_ rules that behave differently depending on PHP runtime, that's why I wanted to discuss it.

The desired behaviour is too have PHP CS Fixer stable across PHP runtime versions once the codebase is fixed for the highest supported version. This is quite working for now for ex. with attributes which are ignored in lower PHP versions.

If we really want `sprintf` to condition based on PHP version, we will have to add it for PHP >=8.4 version and allow both variants below such version if configured so.

If this is too complicated, we must either always fix it or not.

Reply:
there is no harm of applying `\` on PHP7 etc, apply it regardless of runtime.